libs/platform/src/lib/components/form/step-input/base.step-input.ts

```diff
@@ -0,0 +1,477 @@
+
+        this._calculateCanDecrementIncrement();
+
+        this._updateViewValue();
+
+        this.lastEmittedValue = this._value;
+
+        this._listenToFormErrorState();
+
+        this._listenToAlign();
+    }
+
+    /**@hidden
+     * Override writeValue method to keep input view value up to date
+     */
+    writeValue(value: number): void {
+        super.writeValue(value);
+        this._updateViewValue();
+    }
+
+    /** Increase value */
+    increase(step = this._getStepValueForIncrease()): void {
+        if (!this.canIncrement) {
+            return;
+        }
+        const value = addAndCutFloatingNumberDistortion(this._value, step);
+        this.value = this._validateValueByLimits(value);
+        this._emitChangedValue();
+    }
+
+    /** Decrease value */
+    decrease(step = this._getStepValueForDecrease()): void {
+        if (!this.canDecrement) {
+            return;
+        }
+        const value = addAndCutFloatingNumberDistortion(this._value, -step);
+        this.value = this._validateValueByLimits(value);
+        this._emitChangedValue();
+    }
+
+    /** Increase value by large step */
+    largeStepIncrease(): void {
+        const step = this._getStepValueForIncrease() * this.largerStep;
+        this.increase(step);
+    }
+
+    /** Decrease value by large step */
+    largeStepDecrease(): void {
+        const step = this._getStepValueForDecrease() * this.largerStep;
+        this.decrease(step);
+    }
+
+    /**@hidden
+     * Commit new entered value from view.
+     */
+    commitEnteredValue(value: string): void {
+        const parsedValue = this.parseValueInFocusMode(value);
+        let newValue = parsedValue;
+
+        if (newValue !== null) {
+            newValue = this._validateValueByLimits(newValue);
+        }
+
+        const needToUpdateView = this._value === newValue;
+
+        this.value = newValue;
+
+        if (needToUpdateView) {
+            this._updateViewValue();
+        }
+
+        if (this._value !== this.lastEmittedValue) {
+            this._emitChangedValue();
+        }
+    }
+
+    /**@hidden
+     * Indicates when input gets focused
+     */
+    onFocus(): void {
+        super._onFocusChanged(true);
+        // When focus happend by "Tab" key an input field value gets selected by default.
+        // In cases when new_formatted_value !== previous_value the selection gets lost.
+        // So it's needed to track selection before new value rendering and restore it.
+        const { selectionStart, selectionEnd } = this._getValueSelection();
+        this._updateViewValue();
+        if (selectionEnd - selectionStart > 0) {
+            this._setValueSelection(selectionStart, selectionEnd);
+        }
+    }
+
+    /**@hidden
+     * Indicates when input loses focus
+     */
+    onBlur(): void {
+        super._onFocusChanged(false);
+        this._updateViewValue();
+    }
+
+    /** @hidden */
+    detectChanges(): void {
+        this._cd.detectChanges();
```

Comment:
Do you need to force to detect Changes immediately?  Instead of Mar k?

again if you check out the FormField that listens for changes, it does trigger detect/mark once for its input control

Reply:
Removed this method